patroni/postgresql/__init__.py

```diff
@@ -801,6 +801,15 @@ def remove_data_directory(self):
                         logger.info('Removing WAL directory: %s', pg_wal_realpath)
                         shutil.rmtree(pg_wal_realpath)
 
+                pg_tblsp_path = os.path.join(self._data_dir, 'pg_tblspc')
+                if os.path.exists(pg_tblsp_path):
+                    for tspdn in [os.path.join(pg_tblsp_path, dn) for dn in os.listdir(pg_tblsp_path)
+                                  if parse_int(dn) and os.path.islink(os.path.join(pg_tblsp_path, dn))]:
+                        pg_tsp_path = os.path.realpath(tspdn)
+                        if pg_tsp_path is not None and os.path.exists(pg_tsp_path):
+                            logger.info('Removing user defined tablespace directory: %s', pg_tsp_path)
+                            shutil.rmtree(pg_tsp_path, ignore_errors=True)
+
```

Comment:
This change is irrelevant to max timelines, please move it into a separate PR.
Also using the list comprehension is a bit inappropriate in the for loop, because later you are doing additional checks anyway.

Reply:
Sure. Will submit new PR for this change.